### ISSUE TITLE
HHKB support

### ISSUE THREAD

**Issue by romain-lavoix** [rbreaves/kinto#822]
Hello,

thank you for this awesome tool that has allowed me to switch between MacOS and Ubuntu with much less pain.
However, I actually use it with a tenkeyless keyboard, but I also own a Happy Hacking Keyboard, that I use only on MacOS for now.
Is there any support for HHKB ? Or any tweaks that have been tested ?

Thanks a lot !

**Comment by RedBearAK** [comment]
@romain-lavoix 

(I'm not the Kinto dev, just a user and minor contributor.) 

> any support for HHKB

That looks like it's basically an "Apple" type keyboard, aside from the Ctrl key being in the CapsLock position. The keymapper behind the scenes doesn't care where the keys are located, just what the key codes are. 

The main thing that matters is the key next to the space bar becomes RIGHT_CTRL, which is then used to do what you would normally expect the Cmd key to do. This makes sense because so many shortcuts in Linux are based around the Ctrl key. The Alt key would stay Alt, on an "Apple" type keyboard. That should all be the same on the HHKB as it is on any other Mac keyboard. 

What might be a little odd: On an "Apple" type keyboard, the Ctrl key becomes Meta/Super/Win (Cmd) in all non-terminal apps, and stays LEFT_CTRL in all terminal apps. This normally works out fine. So the main question might be when and where do you rely on that key in the CapsLock position actually being Ctrl? If you only ever use that key as Ctrl in terminals, it should be fine. If you use it as Ctrl in non-terminal apps, that could be weird. Because it will be Meta/Super/Win (Cmd), which won't necessarily do what you expect in Linux apps. 

In short it kind of depends on exactly how you use that keyboard with macOS, and how you'll be using it in Linux. You can always try it and see. I suspect it would mostly work out fine, but it's hard to say. 

If you need the ability to use a Wayland login session (on GNOME or KDE Plasma), or want to use more than one keyboard type at the same time, you'd have to try my offshoot of Kinto that supports those features, among other things: 

https://toshy.app
